google/cloud/bigtable/rpc_retry_policy.h

```diff
@@ -99,28 +121,18 @@ class LimitedTimeRetryPolicy : public RPCRetryPolicy {
   LimitedTimeRetryPolicy();
   template <typename duration_t>
   explicit LimitedTimeRetryPolicy(duration_t maximum_duration)
-      : maximum_duration_(std::chrono::duration_cast<std::chrono::milliseconds>(
-            maximum_duration)),
-        deadline_(std::chrono::system_clock::now() + maximum_duration_) {}
+      : impl_(maximum_duration) {}
 
   std::unique_ptr<RPCRetryPolicy> clone() const override;
   void setup(grpc::ClientContext& context) const override;
   bool on_failure(grpc::Status const& status) override;
-  bool can_retry(grpc::StatusCode code) const override;
 
  private:
-  std::chrono::milliseconds maximum_duration_;
-  std::chrono::system_clock::time_point deadline_;
+  using Impl = google::cloud::internal::LimitedTimeRetryPolicy<grpc::Status,
```

Comment:
Is the shorthand `Impl` used anywhere else?

Reply:
Nope, but the declaration looks very funny otherwise (after clang-format):

```C++
 private:
  google::cloud::internal::LimitedErrorCountRetryPolicy<grpc::Status,
                                                        SafeGrpcRetry>
      impl_;
```

I think the alias makes the intent clearer and the code more readable.

Reply:
Happy to change it if you disagree after looking at the alternative.